PackageExplorer/Interop/NativeDragDrop.cs

```diff
@@ -0,0 +1,315 @@
+                            var iStreamStat = new System.Runtime.InteropServices.ComTypes.STATSTG();
+                            iStream.Stat(out iStreamStat, 0); // this will throw for folders
+
+                            //wrapped the IStream in a Stream
+                            return new IStreamWrapper(iStream, iStreamStat);
+
+                        case TYMED.TYMED_HGLOBAL:
+                            var stream = windowsObjectData.GetData(FileContents);
+
+                            return stream as Stream;
+                    }
+                }
+            }
+            catch { }
+
+            return null;
+        }
+
+        private class IStreamWrapper : Stream
+        {
+            private readonly IStream _inner;
+            private System.Runtime.InteropServices.ComTypes.STATSTG? _stats;
+
+            public IStreamWrapper(IStream istream)
+            {
+                _inner = istream;
+            }
+
+            public IStreamWrapper(IStream istream, System.Runtime.InteropServices.ComTypes.STATSTG stats)
+            {
+                _inner = istream;
+                _stats = stats;
+            }
+
+            private System.Runtime.InteropServices.ComTypes.STATSTG GetStats()
+            {
+                if (_stats == null)
+                {
+                    var stats = new System.Runtime.InteropServices.ComTypes.STATSTG();
+                    _inner.Stat(out stats, 0);
+
+                    _stats = stats;
+                }
+                return _stats.Value;
+            }
+
+            public override bool CanRead => true;
+
+            public override bool CanSeek => true;
+
+            public override bool CanWrite => true;
+
+            public override long Length => GetStats().cbSize;
+
+            public override long Position
+            {
+                get => throw new NotImplementedException();
+                set => throw new NotImplementedException();
+            }
+
+            public override void Flush()
+            {
+                _inner.Commit(0); // https://msdn.microsoft.com/en-us/library/windows/desktop/aa380320%28v=vs.85%29.aspx?f=255&MSPPError=-2147217396
+            }
+
+            public override int Read(byte[] buffer, int offset, int count)
+            {
+                var ptr = Marshal.AllocHGlobal(Marshal.SizeOf<ulong>());
+                try
+                {
+                    if (offset > 0)
+                    {
+                        buffer = buffer.Skip(offset).ToArray();
+                    }
+
+                    _inner.Read(buffer, count, ptr);
+
+                    return (int)Marshal.ReadInt64(ptr);
+                }
+                finally
+                {
+                    Marshal.FreeHGlobal(ptr);
+                }
+            }
+
+            public override long Seek(long offset, SeekOrigin origin)
+            {
+                var ptr = Marshal.AllocHGlobal(Marshal.SizeOf<ulong>());
```

Comment:
Since you are going unsafe... 😉

This could just use stack memory (stack is also fixed memory): 
```csharp
public unsafe override long Seek(long offset, SeekOrigin origin)
{
    long result = 0L;
    var ptr = &result;

    _inner.Seek(offset, (int)origin, (IntPtr)ptr);

    return result;
}
```